zcash_client_memory/src/types/block.rs

```diff
@@ -0,0 +1,154 @@
+use std::{
+    cmp::Ordering,
+    collections::{HashMap, HashSet},
+};
+
+use zcash_client_backend::wallet::NoteId;
+use zcash_primitives::{block::BlockHash, consensus::BlockHeight, transaction::TxId};
+use zcash_protocol::memo::MemoBytes;
+
+/// Internal wallet representation of a Block.
+#[derive(Clone, Debug, PartialEq)]
+pub(crate) struct MemoryWalletBlock {
+    pub(crate) height: BlockHeight,
+    pub(crate) hash: BlockHash,
+    pub(crate) block_time: u32,
+    // Just the transactions that involve an account in this wallet
+    pub(crate) _transactions: HashSet<TxId>,
```

Comment:
This is a problematic representation, because it loses transaction index information (i.e. _where_ in the block each transaction is). I guess that is currently not relevant for the in-memory wallet, but I did see transaction indices used in other places, so I presume that means the information gets stored elsewhere?